src/compose/app.ts

```diff
@@ -166,17 +169,29 @@ export class App {
 			),
 		);
 
-		if (
-			steps.length === 0 &&
-			target.commit != null &&
-			this.commit !== target.commit
-		) {
-			steps.push(
-				generateStep('updateCommit', {
-					target: target.commit,
-					appId: this.appId,
-				}),
-			);
+		if (steps.length === 0) {
+			// Update commit in db if different
+			if (target.commit != null && this.commit !== target.commit) {
+				steps.push(
+					generateStep('updateCommit', {
+						target: target.commit,
+						appId: this.appId,
+					}),
+				);
+			} else if (
```

Comment:
Not sure about this else. Couldn't these steps happen at the same time? Why do they need to happen in different iterations?

Reply:
This is a good point, will change

Reply:
Changed `updateCommit` to happen at the same time as `releaseLock` [here](https://github.com/balena-os/balena-supervisor/pull/2234/commits/02476bbc3ae61df1b42ff61f5094d723d61d9dc9#diff-857bb737cdeb3c33e2caa352769d3f96e0e8a48da6fea254eb99bc627732cec1L172-R215)